symbolic-debuginfo/src/dwarf.rs

```diff
@@ -452,8 +453,24 @@ impl<'d, 'a> UnitRef<'d, 'a> {
 
         if let Some(attr) = reference_target {
             return self.resolve_reference(attr, |ref_unit, ref_entry| {
+                // Self-references may have a layer of indircetion. Avoid infinite recursion
+                // in this scenario.
+                match prior_offset {
+                    Some(prior) => {
+                        if self.offset() == ref_unit.offset() && prior == ref_entry.offset() {
+                            return Ok(None);
+                        }
+                    }
+                    None => {}
+                }
```

Comment:
This is better with `if let`.
```suggestion
                if let Some(prior) = prior_offset {
                    if self.offset() == ref_unit.offset() && prior == ref_entry.offset() {
                        return Ok(None);
                    }
                }
```